kingpin/actors/aws/ecs.py

```diff
@@ -406,7 +420,369 @@ def _execute(self):
+        """Create a service.
+
+        Args:
+            service_name: service name to use.
+            family: task definition family.
+            revision: task definition revision.
+            client_token: uuid for the service creation.
+        """
+        create_parameters = dict(
+            cluster=self.option('cluster'),
+            serviceName=service_name,
+            taskDefinition='{0}:{1}'.format(family, revision),
+            desiredCount=self.option('count'),
+            clientToken=client_token,
+            **self.service_definition)
+
+        self.log.info('Creating service.')
+
+        yield self.thread(
+            self.ecs_conn.create_service,
+            **create_parameters)
+
+    @gen.coroutine
+    @dry('Would update service.')
+    def _update_service(self, service_name, family, revision):
+        """Update a service.
+
+        Args:
+            service_name: service name to use.
+            family: task definition family.
+            revision: task definition revision.
+        """
+        deployment_configuration = self.service_definition[
+            'deploymentConfiguration']
+        update_parameters = dict(
+            cluster=self.option('cluster'),
+            service=service_name,
+            taskDefinition='{0}:{1}'.format(family, revision),
+            desiredCount=self.option('count'),
+            deploymentConfiguration=deployment_configuration)
+
+        self.log.info('Updating service.')
+
+        yield self.thread(
+            self.ecs_conn.update_service,
+            **update_parameters)
+
+    @gen.coroutine
+    def _register_service(self, service_name, family, revision):
+        """Registers a service.
+
+        This handles the logic of either:
+            1) Creating a new service if it doesn't exist.
+            or
+            2) Updating the existing service.
+
+        Args:
+            service_name: service_name to use.
+            family: Task Definition family.
+            revision: Task Definition revision.
+        """
+        existing_service = yield self._describe_service(service_name)
+
+        if not existing_service or (existing_service and
+                                    existing_service['status'] == 'INACTIVE'):
+            # Generate a 32 character uuid for the client token.
+            client_token = str(uuid.uuid4())
+            yield self._create_service(service_name,
+                                       family, revision,
+                                       client_token)
+        else:
+            errors = _get_immutable_field_errors(
+                old_params=existing_service,
+                new_params=self.service_definition,
+                immutable_fields=['loadBalancers', 'role'])
+
+            if errors:
+                for error in errors:
+                    self.log.error(error)
+                raise exceptions.UnrecoverableActorFailure(
```

Comment:
This is still a `RecoverableActorFailure` -- its an expected and planned for exception. The user can choose to ignore this and let the rest of their Kingpin script continue if they wish.